packages/headless/src/controllers/folded-result-list/headless-folded-result-list.ts

```diff
@@ -79,7 +84,7 @@ export interface FoldedResultListState extends ResultListState {
   /**
    * The unsorted hierarchical collections of results.
    * */
-  results: FoldedResult[];
+  results: Collection[];
```

Comment:
I'm cautious on introducing a new name. We have `FoldedResult`, `ResultWithFolding` and now `Collection`.

Are three interfaces needed? How can we simplify?
Can we remove `FoldedResult` if it's no longer useful?

Reply:
`ResultWithFolding` is unrelated to the other two, it represents the response from the API and types attributes that we need exclusively for the logic that folds results. This interface doesn't need to be exposed.

`Collection` represents the root result of a collection and contains properties specific to a collection such as whether more results are available in the collection and if more results are loading.

`FoldedResult` represents a result in a folded hierarchy. It's only needed if we expect that we may have more than 2 levels of results, otherwise we could use `Collection` alone.

Reply:
Assuming we wanted to support more than two levels of results, wouldn't the `FoldedResult` need to have properties like `moreResultsAvailable` for the user to expand that portion of the tree?

Why not instead keep the same interface through out the tree, and just set `moreResultsAvailable` and `moreResultsLoading` to false on the children?

Reply:
We can't expand deeper parts of the tree. In order to do so, we would need to be able to detect for each leaf whether there's more children, which would require several requests to be executed for each search.

Exposing `moreResultsAvailable` and giving it `false` would give the impression that no leaf deeper in the hierarchy has other results underneath, which isn't always true. Executing `loadCollection` on its ancestor collection may reveal more results. Without several requests, we can only detect if there's more results in the same collection as a whole.

Reply:
Ok, it sounds like the root result of a collection, and the items folded under it are actually different entities, so having different interface names seems like the correct approach.

However, I would prefer if the names were closer to one another so that the relationship could be clearer. For example,
instead of:

`Collection` and `FoldedResult` which to an outsider would seem like unrelated names, could we have 

`FoldedParentResult` and `FoldedChildResult`? or maybe the child is just a regular `Result`?

If these need to be grouped, then a "collection" would be 1 root and `n` children.

```
interface FoldedCollection {
    parent: FoldedParentResult;
    children: Result[];
}
```

Following, `loadCollection` would only accept a `Result`, and not a `FoldedCollection` or `FoldedParentResult`?

Reply:
I'll move that discussion to the next PR.